lib/generators/deimos/schema_class/templates/schema_record.rb.tt

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+# This file is autogenerated by Deimos, Do NOT modify
+module Deimos
+  # :nodoc:
+  class <%= Deimos::SchemaBackends::AvroBase.schema_classname(@current_schema) %> < SchemaClass::Record
+<%- if @field_assignments.select{ |h| h[:is_schema_class] }.any? -%>
+    # Attribute Readers
+<% end -%>
+<%- @field_assignments.select{ |h| h[:is_schema_class] }.each do |method_definition| -%>
+    # @return [<%= method_definition[:deimos_type] %>]
+    attr_reader :<%= method_definition[:field].name %>
+<% end -%>
+<%- if @field_assignments.select{ |h| !h[:is_schema_class] }.any? -%>
+    # Attribute Accessors
+<% end -%>
+<%- @field_assignments.select{ |h| !h[:is_schema_class] }.each do |method_definition| -%>
+    # @param <%= method_definition[:method_argument] %> [<%= method_definition[:deimos_type] %>]
+    attr_accessor :<%= method_definition[:field].name %>
+<% end -%>
+<%- unless @schema_is_key -%>
+    # @return [Object] An optional payload key
+    attr_accessor :payload_key
+<% end -%>
+<%- if @field_assignments.select{ |h| h[:is_schema_class] }.any? -%>
+    # Attribute Writers
+<% end -%>
+<%- @field_assignments.select{ |h| h[:is_schema_class] }.each do |method_definition| -%>
+    # @param <%= method_definition[:method_argument] %> [<%= method_definition[:deimos_type] %>]
+    def <%= method_definition[:field].name %>=(<%= method_definition[:method_argument] %>)
+<%- if method_definition[:field_type] == :array -%>
+      @<%= method_definition[:field].name %> = values.map do |value|
+        <%= method_definition[:field_initialization] %>
+      end
+<%- elsif method_definition[:field_type] == :map -%>
+      @<%= method_definition[:field].name %> = values.transform_values do |value|
+        <%= method_definition[:field_initialization] %>
+      end
+<%- else -%>
+      @<%= method_definition[:field].name %> = <%= method_definition[:field_initialization] %>
+<% end -%>
+    end
+
+<% end -%>
+
+    # @override
+    <%= @initialization_definition %>
+      super()
```

Comment:
remove `()`?

Reply:
doesn't work without `()` :( I think it's because we're using Keyword args and technically there are **no** args

```ruby
     Failure/Error: super

     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```

Reply:
Found a related article - https://www.rubyguides.com/2018/09/ruby-super-keyword/

> But what if the parent method doesn't take the same number of arguments?
In that case you can use:
`super()` for no arguments
`super(arg1, arg2, ...)` to choose what arguments you want to pass

Makes me wonder if we _should_ include `super()` here at all - the base classes (record, enum and base) don't do any special initialization _yet_

Reply:
If `super` fails then we're defining these classes wrong somehow - either the base or the child classes have the method defined differently.

Reply:
hrm - will try something out :) - gonna define empty `initialize`'s for everything

Reply:
so this only works if i have the following in my base class
```ruby
      # :nodoc:
      def initialize(_)
      end
```

i need to manually define the initialize method with a single argument `_` before i can change `super()` to `super`

Reply:
@dorner tagging for visibility! i can make the change mentioned above for `def initialize(_)` but am unsure if that's proper definition for a base/child class